### PR TITLE
@joystream/types - PoroposalDetails and WorkingGroup enums update (Babylon)

### PR DESCRIPTION
**Pull request by Lezek123**
Minor changed in `@joystream/types` following last 3 commits in https://github.com/Joystream/joystream/pull/1385:
- Added `Content` variant in `WorkingGroup` enum
- Removed no longer supported proposals from `ProposalDetails` enum
- Ran `generate:all` using the latest node

I additionally implemented `toArray()` method in `JoyBTreeSet` (noticed that the original `BTreeSet` types doesn't have it while working on the CLI)

This partially implements https://github.com/Joystream/joystream/issues/1392
I didn't remove `types/content-working-group` yet to prevent some issues with Pioneer and auto-genrated augmentations (since this module is still exposed by the runtime).

I'll adjust Pioneer in a separate PR (see issue: https://github.com/Joystream/joystream/issues/1394)

### CODE REVIEW

**Comment by mnaamani** [comment]
>     * Removed no longer supported proposals from `ProposalDetails` enum

Actually removing the variants from the enum in the runtime here: https://github.com/Joystream/joystream/pull/1385/files#diff-d826bd63d452cdb624a717e8355d24a3L58

Is not safe especially since this will be a runtime upgrade, because it will change values of the variants following it and this will cause issues reading state before the runtime upgrade. They should be kept and just a code comment left like for:

```
    /// ********** Deprecated during the Nicaea release.
    /// It is kept only for backward compatibility in the Pioneer. **********
    /// AccountId for the `evict storage provider` proposal
    EvictStorageProvider(AccountId),
```

**Comment by Lezek123** [comment]
> Actually removing the variants from the enum in the runtime here (...) is not safe

I agree that this could cause issues by generating one of those cases where reading old chain state will become impossible unless using old types and metadata (though I believe the new version of `@polkadot/api` implements some mechanism of switching types&metadata based on which block is beeing queried, which could help in situations like this).

It would also require some tweaks for `Historical proposals` in Pioneer, since they "mock" the `ProposalDetails` type based on json input and this will not work if some enum variants no longer exist.

**Comment by mnaamani** [comment]
So just a correction, with respect to the runtime, after checking the code, in its current form it is "safe" to change the variant ordering (as long as the runtime doesn't read the map that stores the ProposalDetails value). It is still an issue for pioneer.

If it is not in scope now to update pioneer, cli and historical proposals to deal with this, then we can just add back those variants. I prepared a PR here https://github.com/Joystream/joystream/pull/1510

File: types/src/proposals.ts [hunk 1]
```diff
@@ -254,8 +250,6 @@ export class ProposalDetails extends JoyEnum({
   RuntimeUpgrade: Bytes,
   SetElectionParameters: ElectionParameters,
   Spending: SpendingParams,
-  SetLead: SetLead,
-  SetContentWorkingGroupMintCapacity: u128,
```
Comment (mnaamani) [inline types/src/proposals.ts:258]: So I guess we will keep these and rename to with `Deprecated` prefix. As per 

https://github.com/Joystream/joystream/pull/1510/files#diff-d826bd63d452cdb624a717e8355d24a3R58

And same for other deprecated variants: `EvictStorageProvider`, `SetStorageRoleParameters`

**Comment by mnaamani** [comment]
> I'll adjust Pioneer in a separate PR (see issue: #1394)

I guess that explains why the pioneer check is failing.

Otherwise changes look good.

**Pull request merged**